src/java/main/org/apache/zookeeper/server/quorum/QuorumPeer.java

```diff
@@ -1451,7 +1452,12 @@ public synchronized void restartLeaderElection(QuorumVerifier qvOLD, QuorumVerif
     }
 
     public String getNextDynamicConfigFilename() {
-        return configFilename + QuorumPeerConfig.nextDynamicConfigFileSuffix;
+        if (configFilename != null) {
```

Comment:
A few comments:
- It's be better if configFileName is never null. But that may be a bigger change that we don't want to do here
- If its null, it may be better to return null here and avoid calling writeDynamicConfig. 
- The calling test should either initiate configFileName properly or not rely on its existance to check reconfig completion. You can do the latter by looking at the quorumVerified directly.